src/components/ModalBackupWords.js

```diff
@@ -228,87 +257,126 @@ class ModalBackupWords extends React.Component {
+            {renderValidationStep(validationStep)}
+          </div>
         </div>
-      )
-    }
+      );
+    };
 
     const renderBody = () => {
       if (this.props.needPassword && !this.state.passwordSuccess) {
         return renderAskPassword();
       } else if (this.props.words && !this.state.showValidation) {
         return renderShowWords();
       } else if (this.state.showValidation) {
-        return renderValidateBackup();
+        return renderValidation();
       }
-    }
+    };
 
     return (
-      <div className="modal fade" id="backupWordsModal" tabIndex="-1" role="dialog" aria-labelledby="backupWordsModal" aria-hidden="true">
-        <div className="modal-dialog" role="document">
-          <div className="modal-content">
-            <div className="modal-header">
-              <h5 className="modal-title" id="exampleModalLabel">{t`Backup Words`}</h5>
-              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
-                <span aria-hidden="true">&times;</span>
+      <div
+        className='modal fade'
+        id='backupWordsModal'
+        tabIndex='-1'
+        role='dialog'
+        aria-labelledby='backupWordsModal'
+        aria-hidden='true'
+      >
+        <div className={`modal-dialog ${this.state.showValidation ? 'show-validation' : ''}`} role='document'>
+          <div className='modal-content'>
+            <div className='modal-header'>
+              <h5
+                className='modal-title'
+                id='exampleModalLabel'
+              >{t`Backup Words`}</h5>
+              <button
+                type='button'
+                className='close'
+                aria-label='Close'
+                onClick={this.handleClose}
+              >
+                <span aria-hidden='true'>&times;</span>
               </button>
             </div>
-            <div className="modal-body">
+            <div className='modal-body'>
               {renderBody()}
-              <div className="row mt-3">
-                <div className="col-12 col-sm-10">
-                    <p className="error-message text-danger">
-                      {this.state.errorMessage}
+              <div className='row'>
+                <div className='col-12'>
+                  {this.state.errorMessage && (
+                    <p className='error-message text-danger'>
+                      <i className='fa fa-times-circle'/> {this.state.errorMessage}
                     </p>
+                  )}
                 </div>
               </div>
+              {this.state.showValidation && (
+                renderValidationProgress()
+              )}
             </div>
-            <div className="modal-footer">
-              {renderButtons()}
-            </div>
+            {!this.state.showValidation && (
+              <div className='modal-footer'>{renderButtons()}</div>
+            )}
           </div>
         </div>
       </div>
-    )
+    );
   }
 }
 
```

Comment:
I'll add the comment here so we can discuss it in a thread.

Currently, there's one modal that displays the words and another that handles the backup. And when we close the second one the first is displayed. I'd prefer if there's only one modal for everything.

So the words are displayed in a modal and the backup process happens on the same one. When someone makes a mistake and starts over, he could be sent to the words list again, before restarting the backup process.

What do you think? @thr0wn @pedroferreira1

Reply:
@obiyankenobi, technically the words and the backup/validation are displayed in the same modal, only its content is changing (`this.state.showValidation` controls which one is displayed). It may seem like two models due to the width change on the backup/validation:

<table>
  <thead>
    <tr>
      <th>Words</td>
      <th>Validation</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
<img width="811" alt="image" src="https://user-images.githubusercontent.com/5104565/100812298-706ec100-341b-11eb-98a3-dd8d0254688d.png">
      </td>
      <td>
        <img width="816" alt="image" src="https://user-images.githubusercontent.com/5104565/100812263-5c2ac400-341b-11eb-9835-fb17250416a2.png">
      </td>
    </tr>
  </tbody>
</table>

With no changes on the modal width and no close button on the backup validation, it should behave as you suggested.
What do you think?

Reply:
What gave that impression is that, on the first modal (Selecting the wrong word), when you press close, it displays the words again.

And probably keeping the same width is better. You can even increase the width size on the words step if it's better.

Reply:
I agree that keeping the same width is better. I don't really like the width of the "new modal", I would decrease it a bit and the other can increase to match.

I will give my impressions on the UI here in this thread.

1. I think the `handleClose` method shouldn't move back to the words modal, doing that we need to click 'close' twice to close the modal. So `handleClose` should change the state `showValidation` back to false and close the modal (you should change the state only after modal hidden event is triggered)
2. I think the validation progress should show the step you are in (we 'paint' the circle purple only after you choose the word, I think the progress should say 'you are in step 1', so 1 purple circle painted). - I am not sure about this suggestion, I would like to hear your thoughts by my instinct says it's better this way.
3. I know it's not from this PR but I believe the success message is being shown really fast. To update it you can go to `src/screens/NewWallet.js` line 123 and update from 1000 to 3000 maybe.

What do you think? @obiyankenobi @thr0wn

Reply:
Compiling all suggestions, we have the following:

1. Success
![back-success](https://user-images.githubusercontent.com/5104565/101055250-8342dc00-3568-11eb-9b22-7f1ccea47061.gif)

2. Error
![back-error](https://user-images.githubusercontent.com/5104565/101055363-96ee4280-3568-11eb-965f-ec370dbf2cb0.gif)

3. Close
![backup-close](https://user-images.githubusercontent.com/5104565/101055408-a1a8d780-3568-11eb-968f-5cd90767a331.gif)

Sounds good?

Reply:
Looks good.

Reply:
Yeah, I like it